src/cmd/cmd.zig

```diff
@@ -558,29 +760,54 @@ fn accountsDb(_: []const []const u8) !void {
     );
     defer allocator.free(accounts_path);
 
-    var accounts_path_exists = true;
-    std.fs.cwd().access(accounts_path, .{}) catch {
-        accounts_path_exists = false;
+    var maybe_snapshot_files: ?SnapshotFiles = SnapshotFiles.find(allocator, snapshot_dir_str) catch |err| blk: {
+        // if we cant find the full snapshot, we try to download it
+        if (err == error.NoFullSnapshotFileInfoFound) {
+            break :blk null;
+        } else {
+            return err;
+        }
     };
-    const should_unpack_snapshot = !accounts_path_exists or force_unpack_snapshot;
 
-    var snapshot_files = try SnapshotFiles.find(allocator, snapshot_dir_str);
+    const should_download_new_snapshot = force_new_snapshot_download or maybe_snapshot_files == null;
+    if (should_download_new_snapshot) {
+        const trusted_validators = try getTrustedValidators(gpa_allocator);
+        defer if (trusted_validators) |*tvs| tvs.deinit();
+
+        const min_mb_per_sec = min_snapshot_download_speed_mb_option.value.int.?;
+        try downloadSnapshotsFromGossip(
+            allocator,
+            logger,
+            trusted_validators,
+            gossip_service orelse return error.SnapshotsNotFoundAndNoGossipService,
+            snapshot_dir_str,
+            @intCast(min_mb_per_sec),
+        );
+        maybe_snapshot_files = try SnapshotFiles.find(allocator, snapshot_dir_str);
+    }
+
+    // SAFE: this unwrap is safe because we either found the snapshot or downloaded it
+    var snapshot_files = maybe_snapshot_files.?;
```

Comment:
great point - c46360f

Reply:
FYI, In zig 0.12 it becomes a hard error to make something a var when it can be const. This is going to impact many lines of code when we upgrade. I don't think you need to change anything here but I'm just mentioning it because I see you used var for maybe_snapshot_files.

I used to feel that curly braces should *always* be used with if statements, to provide clarity and to avoid mistakes. But auto-formatters and linters can do almost the same thing. Also there are some cases in zig where omitting braces has readability advantages, like avoiding the whole block labeling + `break` pattern.